packages/shared/src/pipelines.ts

```diff
@@ -0,0 +1,95 @@
+export interface Operation<T, D = unknown> {
+  invoke(data: D): T;
+}
+
+export class Pipeline<T, D> implements Operation<T[]> {
+  private operations: Operation<T>[] = [];
+
+  addOperation(operation: Operation<T>) {
+    this.operations.push(operation);
+    return this;
+  }
+
+  invoke(data: D): T[] {
+    return this.operations.map(operation => operation.invoke(data));
+  }
+}
+
+export interface IOOperation<T, D = unknown, O = unknown> {
+  get(): T;
+  set(data: D, options: O): void;
+  remove(): void;
+}
+
+export class IOPipeline<T, D, O = unknown> implements IOOperation<T[]> {
+  private operations: IOOperation<T>[] = [];
+
+  addOperation(operation: IOOperation<T>) {
+    this.operations.push(operation);
+    return this;
+  }
+
+  get(): T[] {
+    return this.operations.map(operation => operation.get());
+  }
+
+  set(data: D, options: O): void {
+    this.operations.map(operation => operation.set(data, options));
+  }
+
+  remove(): void {
+    this.operations.map(operation => operation.remove());
+  }
+}
+
+const getCookieSuffixFromPk = (publishableKey: string): string => {
```

Comment:
I wouldn't expect these cookie-specific methods to live in this file, just the pipeline primitives. I see the IOOperations are defined in shared/cookies, maybe that would be a better place for these other operations as well.

Reply:
i agree, it should be moved to a more suited place. This is a draft PR, If we agree on the implementation, i will move them to a more appropriate location.

Reply:
i have moved any code related to cookie suffixes to the `@clerk/shared/keys`